File: DATASTRUCTURES/Stack.js
 function stack() {
    let array = [];
    let top = 0

    function push(value) {
        array.push(value);
        top++;
    }

    function pop() {
        return array[--top];
    }

    function peek() {
        return array[top - 1];
    }

    function isEmpty() {
        if (top == 0) {
            return true;
        }
        return false;
    }

    function size() {
        return top;
    }

    function clear() {
        let element = []; 
        return element;                                                                
    }

    function value() {
        return array;
    }

    return {
        push,
        pop,
        peek,
        isEmpty,
        size,
        clear,
        value
    }

}
module.exports = stack;

File: DS/Stack.java
public class Stack {
    public static void main(String[] args) {
       
    int[] array = new int[100];
    int top = 0;

    }

    public static int push(int value) {
        array.push(value);
    }

    public void pop() {
        return array[--top];
    }

    public void peek() {
        return array[top - 1];
    }

    public void isEmpty() {
        if(top == 0) {
            return true;
        }
        return false;
    }

    public void size() {
        return top;
    }

    public clear() {
        int element = [];
        return element;
    }

    public void value() {
        return array;
    }
}